conf/mpidistutils.py

```diff
@@ -908,13 +908,38 @@ def configure_mpi(ext, config_cmd):
       return 0;
     }
     """)
-    errmsg = "Cannot {} MPI programs. Check your configuration!!!"
+    errmsg = [
+        "Cannot {} MPI programs. Check your configuration!!!",
+        "Installing mpi4py requires a working MPI implementation.",
+    ]
+    if sys.platform == 'linux':
+        errmsg += [
+            "You may be missing the MPICH or Open MPI devel package:",
+            "* On Fedora/RHEL:",
+            "  $ sudo dnf install mpich-devel     # for MPICH",
+            "  $ sudo dnf install openmpi-devel   # for Open MPI",
+            "* On Debian/Ubuntu:",
+            "  $ sudo apt install libmpich-dev    # for MPICH",
+            "  $ sudo apt install libopenmpi-dev  # for Open MPI",
+        ]
```

Comment:
```suggestion
            "  $ sudo apt install libopenmpi-dev  # for Open MPI",
            "* Or check with your system administrator / refer to the user guide if using a cluster."
        ]
```

Reply:
And Leo Fang just proved my point... whatever you write, is never enough, there is always something extra to say...

Reply:
(Suggestion updated.)

Reply:
@leofang But your suggestion is wrong, you are missing the `{mpich|openmpi}-mpicc` package to pull in the compiler package, otherwise things will not work. Let me try to address it.

Reply:
IIRC didn't we have bug reports for the mpicc wrappers from conda-forge?

Reply:
Maybe, but I don't recall the context exactly. The fact is, the `mpicc` compiler wrapper and friends have the compiler command hardwired to the one used to build the MPI package. The <mpi>-mpicc packages are just convenience matapackages to install MPI + compiler toolchain. Without that, users get a confusing "command not found" error message.

Reply:
IIRC the problem was that conda-forge compiler does not play well, and if we bring in compiler toolchain from conda, instead of using the system one and just manually setting up `-I` and `-L`, we could run into issues.

Reply:
So let's remove all references to conda then. I don't want a clarification that contributes to spread even more confusion.